food-truck-frontend/src/components/truck/TruckListAndMapComponent.tsx

```diff
@@ -42,7 +43,31 @@ function TruckListAndMapComponent(props: Props) {
         </Box>
         <TruckLocationMapComponent locations={props.routePts} height="50vh"/>
       </GridListTile>
+
+        { props.recommendedTrucks !== undefined ?
+
+            <GridListTile cols={2} style={{height: '50vh'}}>
+                <Box pt={1} px={3}>
+                    <Typography variant="h6">Recommended</Typography>
+                </Box>
+                <Container style={{maxHeight: '50vh', overflow: 'auto'}}>
+                    <List disablePadding>
+                        {
+                            props.recommendedTrucks.map(truck => (
+                                <ListItem key={truck.first.id} style={{minWidth: '100%'}} disableGutters>
+                                    <TruckCardComponent id={truck.first.id}
+                                                        userOwnsTruck={props.owner !== undefined ? props.owner : false}/>
```

Comment:
This is incorrect usage of the `owner` flag. This just indicates that the user is an owner account, but not necessarily that they own a specific truck id. You need to compare it with the truck's `userId`.

Reply:
Yep, I decided to change it as false.